newdle/core/util.py

```diff
@@ -33,3 +36,24 @@ def range_union(ranges):
         else:
             results.append((start, end))
     return results
+
+
+def sign_user(user):
+    """Sign user data (UID + e-mail)."""
+    signer = Signer(current_app.config['SECRET_KEY'], salt='newdle-users')
+    email = user['email']
+    uid = user['uid']
+    return dict(
+        user,
+        signature=signer.get_signature(f'{uid}-{email}'.encode('ascii')).decode(
```

Comment:
shouldn't we include the name as well? (maybe by using `frozenset(user.items())` as input)

Reply:
`frozenset(...)`? Why? We need to ensure consistent sorting between attributes. I was thinking more of something like:

```python
'-'.join(v for k, v in sorted(user.items()))
```

Reply:
I was hoping the signer would automatically take care of converting non-string data to a suitable format ;) but yeah, if it has to be a string then sorting manually is needed of course..